rafal_szyinski/main.cpp

```diff
@@ -43,17 +43,15 @@ void loop()
 			mode_button_state = reading;
 			if (mode_button_state == HIGH)
 			{ 
-				turn_off_color();
-				state = state ? static_cast<States>(state + 1) : DISABLED;
-				if (state >= UNKNOW)
-				{
-					state = LEDR;
-				}
+				
```

Comment:
It looks very complicated. It was my bad to force you to merge these if statements. :D
Let's leave this but I would do this in that way.
```cpp
if (mode_button_state == HIGH and state != DISABLED)
{ 
    state = states + 1 > DISCO ? LEDR : static_cast<States>(state + 1);
}
```